test/exceed_test.gd

```diff
@@ -0,0 +1,279 @@
+		for i in range(to_discard):
+			cards.append(player.hand[i].id)
+		assert_true(game_logic.do_discard_to_max(player, cards))
+
+func validate_gauge(player, amount, id):
+	assert_eq(len(player.gauge), amount)
+	if amount == 0 or len(player.gauge) != amount:
+		return
+	assert_true(
+		player.gauge.any(func (card): return card.id == id),
+		"Didn't find card %s in gauge." % id)
+
+func validate_discard(player, amount, id):
+	assert_eq(len(player.discards), amount)
+	if amount == 0 or len(player.discards) != amount:
+		return
+	assert_true(
+		player.discard.any(func (card): return card.id == id),
+		"Didn't find card %s in discard." % id)
+
+func process_decisions(player, strike_state, decisions):
+	while game_logic.game_state == Enums.GameState.GameState_PlayerDecision and \
+			game_logic.active_strike.strike_state == strike_state and \
+			game_logic.decision_info.player == player.my_id:
+		var content = decisions.pop_front()
+		if content == null:
+			fail_test("Player %s needed to decide on %s during %s but wasn't told how to" % [
+					player.my_id, Enums.DecisionType.keys()[game_logic.decision_info.type],
+					LocalGame.StrikeState.keys()[strike_state]])
+			return
+		match game_logic.decision_info.type:
+			Enums.DecisionType.DecisionType_ForceForEffect:
+				assert_true(game_logic.do_force_for_effect(player, content, false),
+						"%s failed to perform a Force effect using %s" % [player, content])
+			Enums.DecisionType.DecisionType_GaugeForEffect:
+				assert_true(game_logic.do_gauge_for_effect(player, content),
+						"%s failed to perform a Gauge effect using %s" % [player, content])
+			Enums.DecisionType.DecisionType_PayStrikeCost_Required, Enums.DecisionType.DecisionType_PayStrikeCost_CanWild:
+				# There is sometimes an init_extra_cost here
+				# TODO: See if anyone needs it to be more than 0
+				assert_true(game_logic.do_pay_strike_cost(player, content, false),
+						"%s failed to pay a Strike cost using %s" % [player, content])
+			var decision_type:  # Unknown decision type, just roll with it
+				assert_true(game_logic.do_choice(player, content),
+						"Decision of type %s unhandled by test harness (attempted by player %s, content %s)" % [
+								decision_type, player, content])
+
+func process_remaining_decisions(initiator, defender, init_choices, def_choices):
+	var empty_loop_count = 0
+	while init_choices.size() + def_choices.size() >= 1:
+		# empty_loop_count is used to detect when there are still prescribed choices in
+		# the input, but the game is not actually making additional decision points available.
+		empty_loop_count += 1
+		if empty_loop_count >= 3:
+			fail_test("Game is not providing decision points to process initiator" +
+					" choices %s or defender choices %s" % [init_choices, def_choices])
+			return
+		while game_logic.game_state == Enums.GameState.GameState_PlayerDecision:
+				## TODO: Figure out if it's really necessary to limit ourselves to this decision type
```

Comment:
what do you mean by "this decision type"? the game state?

Reply:
Ah, yeah, the comment should say game state.

The comment as a whole is there because this function replaces `handle_simultaneous_decisions` (see any of the unmodified test files), which specifies that it only cares about that that game state + `DecisionType_ChooseSimultaneousEffect`, and I'm not sure how many or which sorts of things fall under the umbrella of that decision type as opposed to one of the others.